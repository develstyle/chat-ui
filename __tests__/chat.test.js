import React, { Component } from 'react';

import LocalStorage from '../__mocks__/localStorage';
import Transport from '../__mocks__/transport';


import Chat  from '../src/components/chat/index';
import { shallow } from 'enzyme';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';


describe('Test chat', () => {


    describe('Test render components', () => {

        let transport;

        beforeEach(() => {
            transport = new Transport();
        })

        it('render a chat', () => {
            const mounted = shallow(<Chat transport={transport} storage={LocalStorage}/>);

            expect(mounted.find('#chat').length).toEqual(1);
            expect(mounted.find('#newMessageBox').length).toEqual(1);
        });

        it('render and check snapshot', () => {
            const rendered = renderer.create(
                <MuiThemeProvider>
                    <Chat transport={transport} storage={LocalStorage}/>
                </MuiThemeProvider>
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });

    });

    describe('Test chat business logic', () => {

        let transport;
        let chat;

        beforeEach(() => {
            transport = new Transport();
            chat = new Chat({transport: transport, storage: LocalStorage});
            chat.setState = function (state) {
                this.state = state;
            }
        });

        it('test adding messages to list', () => {
            expect(chat.state.messages.length).toEqual(0);
            chat.addMessageToList({text: 'test'});
            expect(chat.state.messages.length).toEqual(1);
        });

        it('test get avatar', () => {
            const avatars = chat.getAvatarsList();
            const avatar = chat.getRandomAvatar();
            const avatarIndex = avatars.indexOf(avatar);

            expect(avatarIndex).toBeGreaterThanOrEqual(0);
            expect(avatarIndex).toBeLessThan(avatars.length);

            const avatar2 = chat.getRandomAvatar();
            expect(avatar2).toEqual(avatar);
        });

    });

});