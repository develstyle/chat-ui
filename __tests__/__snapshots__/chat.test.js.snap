// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test chat Test render components render and check snapshot 1`] = `
<div
  id="chat"
>
  <div
    id="messages"
    style={
      Object {
        "muiPrepared": true,
        "padding": "8px 0px 8px 0px",
      }
    }
  />
  <div
    id="newMessageBox"
  >
    <img
      className={undefined}
      size={40}
      src="https://spotim-demo-chat-server.herokuapp.com/avatars/001-snorlax.png"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(188, 188, 188)",
          "borderRadius": "50%",
          "color": "#ffffff",
          "display": "inline-flex",
          "fontSize": 20,
          "height": 40,
          "justifyContent": "center",
          "muiPrepared": true,
          "userSelect": "none",
          "width": 40,
        }
      }
    />
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 48,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 1,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Enter your name
      </div>
      <input
        defaultValue=""
        disabled={false}
        id="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
      />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
    <div
      className={undefined}
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 48,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 12,
            "color": "rgba(0, 0, 0, 0.3)",
            "muiPrepared": true,
            "opacity": 1,
            "position": "absolute",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        Enter your message here
      </div>
      <input
        disabled={false}
        id="message"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value=""
      />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
      
    </div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        id="btnSendMessage"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#00bcd4",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Send message
            </span>
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
